feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,400 @@
+        
+        Implementation
+        --------------
+
+            Private methods
+            ---------------
+                `__target_mean_encoding(self, df_train_, df_test_, var_cols=None)`
+                `__validate_variables(self, X)`
+                    
+            Public methods
+            --------------
+                `fit(self, X, y=None)`
+                `transform(self)`
+                `fit_transform(self, X, y=None)`
+        
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all 
+        variables in the dataset associated with the variables_type.
+
+    target: string, default=None
+        The target variable to evaluate. If None, the transformer will return the 
+        relevant message for the missing value.
+
+    variables_type: boolean, default=True
+        If variables_type = True: the transformer will evaluate the variables as the 
+            categorical variables. 
+        If variables_type = False: the transformer will evaluate the variables as the
+            numerical variables.
+
+    metrics_score: string, default='roc_auc_score'
+        This indicates the metrics score to perform the feature selection.
+        The current support includes 'roc_auc_score' and 'r2_score'.
+
+    test_size: float, default=0.3
+        The test size setting of the data in the train_test_split method.
+
+    random_state: int, default=0
+        The random state setting in the train_test_split method.
+
+    quantiles: int, default=5
+        The amount of quantiles setting in the qcut of the fit method in case of the numerical types.
+        The range of the quantiles values is in between 3 and 100. The transformer will return a message
+        if the input quantiles value is out of this range.
+
+    """
+
+    def __init__(self, variables=None, target=None, variables_type=True, metrics_score='roc_auc_score', test_size=0.3, random_state=0, quantiles=5):
+        # inputs
+        self.variables = _define_variables(variables)
+        self.vars_type = variables_type
+        self.metrics_score = metrics_score
+        self.target = target
+        self.test_size = test_size
+        self.random_state = random_state
+        self.quantiles = quantiles
+
+        # transformed variables
+        self.target_enc_dict = {}
+        self.transformed_cols = []
+        self.binned_vars = []
+        
+        # dataframes
+        self.X_train, self.X_test = None, None
+        self.y_train, self.y_test = None, None
+        self.X_train_enc, self.X_test_enc = None, None
+        
+        # output
+        self.output = None
+
+
+    def __target_mean_encoding(self, df_train_, df_test_, var_cols=None):
+        """
+        Performs
+        --------
+            For each variable:
+                + Performs the target mean encoding on the train set. 
```

Comment:
to determine target mean encoding, we can use the MeanEncoder from feature-engine. Could you experiment with that? I don't think we need to write this functionality from scratch.

If I remember correclty, the MeanEncoder will encode categories with the target Mean, and for numerical variables, they need to be discretised first, making sure we set the return_object=True and then we can run the MeanEncoder at the back

Reply:
Yes, I will use the MeanEncoder instead of the __target_mean_encoding method. Thanks Sole!